packages/core/src/docdb/commands/deleteInstance.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger, ToolkitError } from '../../shared'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+import { DBInstanceNode } from '../explorer/dbInstanceNode'
+import { DBClusterNode } from '../explorer/dbClusterNode'
+import { telemetry } from '../../shared/telemetry'
+
+/**
+ * Deletes a DocumentDB instance.
+ *
+ * Prompts the user for confirmation.
+ * Deletes the instance.
+ * Refreshes the parent cluster node.
+ */
+export async function deleteInstance(node: DBInstanceNode) {
+    getLogger().debug('docdb:DeleteInstance called for: %O', node)
+
+    await telemetry.docdb_deleteInstance.run(async () => {
+        if (!node) {
+            throw new ToolkitError('No node specified for DeleteInstance')
+        }
+
+        const parent = node.parent as DBClusterNode
+        const client = parent.client
+        const instanceName = node.instance.DBInstanceIdentifier ?? ''
+
+        if (node.instance.DBInstanceStatus !== 'available') {
```

Comment:
I see this snippet in at least 3 difference places (here, modifyInstance, rebootInstance, and probably a few more), can we pull it out to a function